tests/db_functions/math/test_floor.py

```diff
@@ -0,0 +1,44 @@
+import math
+from decimal import Decimal
+
+from django.db.models import DecimalField
+from django.db.models.functions import Floor
+from django.test import TestCase
+
+from ..models import DecimalModel, FloatModel, IntegerModel
+
+
+class FloorTests(TestCase):
+
+    def test_decimal(self):
+        DecimalModel.objects.create(n1=Decimal('-12.9'), n2=Decimal('0.6'))
+        obj = DecimalModel.objects.annotate(n1_d=Floor('n1'), n2_d=Floor('n2')).first()
+        self.assertEqual(obj.n1_d, math.floor(obj.n1))
+        self.assertEqual(obj.n2_d, math.floor(obj.n2))
+
+    def test_float(self):
+        FloatModel.objects.create(f1=-27.5, f2=0.33)
+        obj = FloatModel.objects.annotate(f1_d=Floor('f1'), f2_d=Floor('f2')).first()
+        self.assertEqual(obj.f1_d, math.floor(obj.f1))
+        self.assertEqual(obj.f2_d, math.floor(obj.f2))
+
+    def test_integer(self):
+        IntegerModel.objects.create(small=-20, normal=15, big=-1)
+        obj = IntegerModel.objects.annotate(
+            small_d=Floor('small'),
+            normal_d=Floor('normal'),
+            big_d=Floor('big'),
+        ).first()
+        self.assertEqual(obj.small_d, math.floor(obj.small))
+        self.assertEqual(obj.normal_d, math.floor(obj.normal))
+        self.assertEqual(obj.big_d, math.floor(obj.big))
+
+    def test_transform(self):
+        try:
+            DecimalField.register_lookup(Floor)
+            DecimalModel.objects.create(n1=Decimal('5.4'), n2=Decimal('0'))
+            DecimalModel.objects.create(n1=Decimal('3.4'), n2=Decimal('0'))
+            objs = DecimalModel.objects.filter(n1__floor__gt=4)
+            self.assertQuerysetEqual(objs, [Decimal('5.4')], lambda x: x.n1)
```

Comment:
Please change this back to `lambda a: a.n1` for consistency with all of the other tests.